scripts/update/0.5-0.6/cluster.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash
+
+# This script:
+# 1) Appends the label `lokomotive.alpha.kinvolk.io/bgp-enabled=true` in the env file for the nodes
+#    running MetalLB.
+# 2) Update the image tag of Kubelet.
+# 3) Update etcd if it is a controller node.
+
+set -euo pipefail
+
+mode=$1
+
+function run_on_host() {
+  nsenter -a -t 1 /bin/sh -c "${1}"
+}
+
+updated_kubelet_config=false
+
+function update_kubelet_env_file() {
+  readonly kubelet_env="/etc/kubernetes/kubelet.env"
+  readonly kubelet_version="v1.19.4"
+
+  if grep "${kubelet_version}" "${kubelet_env}" >/dev/null; then
+    echo "Kubelet env var file ${kubelet_env} is already updated."
+    return
+  fi
+
+  echo -e "\nUpdating Kubelet env file...\nOld Kubelet env file:\n"
+  cat "${kubelet_env}"
+
+  label=$(grep ^NODE_LABELS "${kubelet_env}")
+  label_prefix="${label::-1}"
+  augmented_label="${label_prefix},lokomotive.alpha.kinvolk.io/bgp-enabled=true\""
+  sed_cmd="s|^KUBELET_IMAGE_TAG.*|KUBELET_IMAGE_TAG=${kubelet_version}|g"
+
+  # Update the label on MetalLB nodes.
+  if [ "${mode}" = "metallb" ]; then
+    sed_cmd="${sed_cmd}; s|^NODE_LABELS.*|${augmented_label}|g"
+  fi
+
+  # Update the kubelet image version.
+  sed "${sed_cmd}" "${kubelet_env}" >/tmp/kubelet.env
+
+  # This copy is needed because `sed -i` tries to create a new file, this changes the file inode and
+  # docker does not allow it. We save changes using `sed` to a temporary file and then overwrite
+  # contents of actual file from temporary file.
+  cat /tmp/kubelet.env >"${kubelet_env}"
+
+  echo -e "\nNew Kubelet env file:\n"
+  cat "${kubelet_env}"
+
+  updated_kubelet_config=true
+}
+
+function update_kubelet_config_file() {
+  if [ "${mode}" != "metallb" ]; then
+    echo "Nothing to do. Not a MetalLB node."
+    return
+  fi
+
+  readonly ckcd="/etc/kubernetes/configure-kubelet-cgroup-driver"
+  readonly newline="providerID: external"
```

Comment:
This should only be done on Packet. Do we check that somehow?

Reply:
Yes, this is not a problem because we list the nodes on the basis of a metallb labels. If the nodes don't have it then we simply don't list it.

AWS nodes shouldn't have those labels hence these changes won't be applied on those nodes.

Reply:
Hm, okay, but then `providerID: external` line should be added to all nodes on Packet, not only to MetalLB nodes...